config_parser.cc

```diff
@@ -198,18 +200,28 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
       config_stack.top()->statements_.back().get()->child_block_.reset(
           new_config);
       config_stack.push(new_config);
+      block_lvl++;
     } else if (token_type == TOKEN_TYPE_END_BLOCK) {
-      if (last_token_type != TOKEN_TYPE_STATEMENT_END) {
+      // in addition to checking statement_end, also need to check block_end
+      if (last_token_type != TOKEN_TYPE_STATEMENT_END && last_token_type != TOKEN_TYPE_END_BLOCK) {
         // Error.
         break;
       }
       config_stack.pop();
+      if(block_lvl-- < 0) {
```

Comment:
By the [Google C++ style guide](https://google.github.io/styleguide/cppguide.html#Conditionals), there should be a space after the `if` like so: `if (condition) { ... }`